src/sig/sig.h

```diff
@@ -63,108 +84,123 @@ struct OQS_SIG {
 	size_t length_public_key;
 	/** The (maximum) length, in bytes, of secret keys for this signature scheme. */
 	size_t length_secret_key;
-	/** The (maximum) length, in bytes, of a signature for this signature scheme. */
-	size_t max_length_signature;
+	/** The (maximum) length, in bytes, of signatures for this signature scheme. */
+	size_t length_signature;
 
-	/**
-	 * Opaque pointer for passing around any computation context
-	 */
-	void *ctx;
-
-	// clang-format and doxygen don't work together nicely on these three functions
 	// clang-format off
+
 	/**
-	 * Pointer to a function for public and private signature key generation.
+	 * Keypair generation algorithm.
+	 *
+	 * Caller is responsible for allocating sufficient memory for `public_key` and
+	 * `secret_key`, based on the `length_*` members in this object or the per-scheme
+	 * compile-time macros `OQS_SIG_*_length_*`.
 	 *
-	 * @param s                The signature structure.
-	 * @param priv             The signer's private key.
-	 * @param pub              The signer's public key.
-	 * @return                 OQS_SUCCESS on success, or OQS_ERROR on failure.
+	 * @param[out] public_key The public key represented as a byte string.
+	 * @param[out] secret_key The secret key represented as a byte string.
+	 * @return OQS_SUCCESS or OQS_ERROR
 	 */
-	OQS_STATUS(*keygen)(const OQS_SIG *s, uint8_t *priv, uint8_t *pub);
+	OQS_STATUS (*keypair)(uint8_t *public_key, uint8_t *secret_key);
 
 	/**
-	 * Pointer to a function for signature generation.
+	 * Signature generation algorithm.
 	 *
-	 * @param s                The signature structure.
-	 * @param priv             The signer's private key.
-	 * @param msg              The message to sign.
-	 * @param msg_len          Length of the message to sign.
-	 * @param sig              The generated signature. Must be allocated by the caller, or NULL to learn how much space is needed, as returned in sig_len.
-	 * @param sig_len          In: length of sig, out: length of the generated signature.
-	 * @return                 OQS_SUCCESS on success, or OQS_ERROR on failure.
+	 * Caller is responsible for allocating sufficient memory for `signed_message`,
```

Comment:
Change documentation to remove `signed_message`.

Also, should we have a comment explicitly explaining that in this branch `signature` contains just the digest, whereas in nist-branch `signature` contains both the digest and the message?

Reply:
Changed. We need to explain it somewhere, but I'm not sure where. For folks using the master branch for integration, it might not make sense to talk about a "nist-branch"; and the signature/verify paradigm used by the master branch is the classic API for signatures.